jax/_src/lax/lax.py

```diff
@@ -2175,21 +2175,43 @@ def _minmax_complex_lowering(x, y, *, lax_cmp_pick_x):
 mlir.register_lowering(shift_right_logical_p,
                        partial(_nary_lower_hlo, hlo.ShiftRightLogicalOp))
 
+def _opaque_comparison_hlo(direction, reduction_op, identity, ctx,
```

Comment:
I wonder if we will need this helper functions in other lowering rules, besides those in `lax.py`. In that case it makes sense to put it in `mlir.py`.